server/src/main/scala/org/scassandra/server/ServerStubRunner.scala

```diff
@@ -52,42 +56,32 @@ class ServerStubRunner( val binaryListenAddress: String = "localhost",
                         val adminListenAddress: String = "localhost",
                         val adminPortNumber: Int = 8043,
                         val startupTimeoutSeconds: Long = 10) extends LazyLogging {
+  import ServerStubRunner.actorSystem
 
-  var system: ActorSystem = _
+  import ExecutionContext.Implicits.global
 
+  // TODO: This is only used by integration tests, move into Actor.
   val primedResults = PrimeQueryStore()
-  val primePreparedStore = new PrimePreparedStore
-  val primePreparedPatternStore = new PrimePreparedPatternStore
-  val primePreparedMultiStore = new PrimePreparedMultiStore
-  val primeBatchStore = new PrimeBatchStore
-  val activityLog = new ActivityLog
-  val preparedLookup = new CompositePreparedPrimeStore(primePreparedStore, primePreparedPatternStore, primePreparedMultiStore)
-
-  var primingReadyListener: ActorRef = _
-  var tcpReadyListener: ActorRef = _
-
-  def start() = {
-    system = ActorSystem(s"CassandraServerStub-$binaryPortNumber-$adminPortNumber")
-    val manager = IO(Tcp)(system)
-    primingReadyListener = system.actorOf(Props(classOf[ServerReadyListener]), "PrimingReadyListener")
-    tcpReadyListener = system.actorOf(Props(classOf[ServerReadyListener]), "TcpReadyListener")
-    val tcpServer = system.actorOf(Props(classOf[TcpServer], manager, binaryListenAddress, binaryPortNumber, primedResults, preparedLookup, primeBatchStore, tcpReadyListener, activityLog), "BinaryTcpListener")
-    system.actorOf(Props(classOf[PrimingServer], adminListenAddress, adminPortNumber, primedResults, primePreparedStore,
-      primePreparedPatternStore, primePreparedMultiStore, primeBatchStore, primingReadyListener, activityLog, tcpServer), "PrimingServer")
-  }
 
-  def awaitTermination() = {
-    system.awaitTermination()
+  var scassandra: ActorRef = _
+
+  def start() = this.synchronized {
```

Comment:
Highly unlikely, it's up to the caller for the most part.  We could do something clever like make this  a typed actor (which will provide a similar API, but manage all the synchronization & lifecycle for us), although not sure whats available with the version of akka we are using.